src/Microsoft.Restier.Publishers.OData/RestierController.cs

```diff
@@ -525,19 +527,25 @@ private IQueryable GetQuery(ODataPath path)
             return queryable;
         }
 
-        private IQueryable ApplyQueryOptions(
-            IQueryable queryable, ODataPath path, bool applyCount, out ETag etag)
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="queryable"></param>
+        /// <param name="path"></param>
+        /// <param name="applyCount"></param>
+        /// <returns></returns>
+            IQueryable queryable, ODataPath path, bool applyCount)
         {
-            etag = null;
+            ETag etag = null;
 
             if (this.shouldWriteRawValue)
             {
                 // Query options don't apply to $value.
-                return queryable;
+                return (queryable, null);
             }
 
             HttpRequestMessageProperties properties = this.Request.ODataProperties();
-            var model = Api.GetModelAsync().Result;
```

Comment:
Actually this won't deadlock since at this time EdmModel must have been ready. It's been ready in WebApiConfig. But of course I welcome this change.